### PR TITLE
feat(relay-kit): Add tx service compatibility to `Safe4337Pack`

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves https://github.com/safe-global/safe-core-sdk/issues/825

## How this PR fixes it
We are adding support for using SafeOperation responses from the transaction service in the `Safe4337Pack`

### CODE REVIEW

File: packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts [hunk 1]
```diff
@@ -476,9 +520,15 @@ export class Safe4337Pack extends RelayKitBasePack<{
    * @param {EthSafeOperation} safeOperation - The SafeOperation to execute.
    * @return {Promise<string>} The user operation hash.
    */
-  async executeTransaction({
-    executable: safeOperation
-  }: Safe4337ExecutableProps): Promise<string> {
+  async executeTransaction({ executable }: Safe4337ExecutableProps): Promise<string> {
+    let safeOperation: EthSafeOperation
+
+    if ('safeOperationHash' in executable) {
```
Comment (tmjssz) [inline packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts:526]: How about defining this as a type guard, so you could use it as `!isSafeOperation(executable)` here, as well as in line 467. You could even use it in [SafeApiKit line 795](https://github.com/safe-global/safe-core-sdk/blob/feat/add-api-compatibility-to-4337-pack/packages/api-kit/src/SafeApiKit.ts#L795)

Comment (yagopv) [reply packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts]: Hey, changed. Defined the guards in the types library together with the type definition

**Review by tmjssz — APPROVED** [review]

File: packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts [hunk 2]
```diff
@@ -415,17 +419,67 @@ export class Safe4337Pack extends RelayKitBasePack<{
...
+
+    if (safeOperationResponse.confirmations) {
+      safeOperationResponse.confirmations.forEach((confirmation: SafeOperationConfirmation) => {
+        safeOperation.addSignature(new EthSafeSignature(confirmation.owner, confirmation.signature))
+      })
+    }
+
+    return safeOperation
+  }
+
```
Comment (dasanra) [inline packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts:461]: Don't you feel we are adding lot of utils like functionality into the main class? We even have some other utils at the bottom of this file that makes the file to be very long.

Comment (yagopv) [reply packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts:461]: yes, you are right

**Review by dasanra — APPROVED** [review]

**Pull request merged**